datajunction-server/datajunction_server/internal/nodes.py

```diff
@@ -912,13 +914,14 @@ async def update_cube_node(
     background_tasks: BackgroundTasks = None,
     validate_access: access.ValidateAccessFn,
     save_history: Callable,
+    refresh_materialization: bool = False,
 ) -> Optional[NodeRevision]:
     """
     Update cube node based on changes
     """
     node = await Node.get_cube_by_name(session, node_revision.name)
     node_revision = node.current  # type: ignore
-    minor_changes = has_minor_changes(node_revision, data)
+    minor_changes = has_minor_changes(node_revision, data) or refresh_materialization
```

Comment:
So in this case, refreshing the materialization will result in a new node revision being created. Is that level of state tracking needed, given that nothing has changed on the node revision itself?